lib/spack/spack/reporters/extract.py

```diff
@@ -153,13 +130,9 @@ def extract_test_parts(default_name, outputs):
                 if msg.startswith("Installing"):
                     continue
 
-                # TODO (post-34236): Remove this check when remove run_test(),
-                # TODO (post-34236): etc. since no longer supporting expected
-                # TODO (post-34236): failures.
-                if msg.startswith("Expecting return code"):
-                    if part:
-                        part["desc"] += f"; {msg}"
-                    continue
+                if part:
+                    part["desc"] += f"; {msg}"
+                continue
```

Comment:
The goal of the original check was to determine whether message processing was done (and append the expectation to the description).  In this case, since the new API does not support expected return codes, it is not.  So the rest of the check needs to be removed.